environment/src/main/java/com/sequenceiq/environment/environment/validation/network/gcp/GcpEnvironmentNetworkValidator.java

```diff
@@ -129,6 +144,38 @@ private static void checkAvailabilityZones(ValidationResult.ValidationResultBuil
         }
     }
 
+    private void checkAvailabilityZones(ValidationResult.ValidationResultBuilder resultBuilder, EnvironmentDto environmentDto, NetworkDto networkDto) {
+        PlatformResourceRequest platformResourceRequest = new PlatformResourceRequest();
+        platformResourceRequest.setCredential(environmentDto.getCredential());
+        platformResourceRequest.setCloudPlatform(environmentDto.getCloudPlatform());
+        CloudRegions regionsByCredential = platformParameterService.getRegionsByCredential(platformResourceRequest, true);
+        Map<Region, List<AvailabilityZone>> cloudRegions = regionsByCredential.getCloudRegions();
+        Region requestedRegion = Region.region(environmentDto.getLocation().getName());
+
+        if (networkDto.getGcp() != null && CollectionUtils.isNotEmpty(networkDto.getGcp().getAvailabilityZones()) && MapUtils.isNotEmpty(cloudRegions)) {
+            Set<AvailabilityZone> requestedZones = networkDto.getGcp().getAvailabilityZones()
+                    .stream()
+                    .map(AvailabilityZone::availabilityZone)
+                    .collect(Collectors.toSet());
+            if (cloudRegions.containsKey(requestedRegion)) {
+                Set<AvailabilityZone> zonesInRegion = new HashSet<>(cloudRegions.get(requestedRegion));
+                if (CollectionUtils.isNotEmpty(zonesInRegion) && !zonesInRegion.containsAll(requestedZones)) {
+                    String msg = String.format(INVALID_ZONE_PATTERN,
+                            requestedRegion.getRegionName(),
+                            requestedZones.stream().map(AvailabilityZone::value).collect(Collectors.joining(",")),
```

Comment:
Since we only allow one zone right now, it is fine. When we allow multiple zones, we will put all the zones in error message even if there are only few which are not valid. We can put only those zones which are invalid instead of putting all the requested zones

Reply:
@adarshsharma1983 all the requested zones there in the error message, it is also unit tested, only the region which is not a collection here :slightly_smiling_face:

Reply:
That's why I am saying that we should not put all requested zones in error message. We should only put those requested zones which are invalid.